garak/probes/donotanswer.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+
+"""
+Do-Not-Answer Probes
+
+https://github.com/Libr-AI/do-not-answer
+https://arxiv.org/abs/2308.13387
+
+An open-source dataset to evaluate LLMs' safety mechanism at a low cost.
+The dataset consists only of prompts to which responsible language models should not answer.
+"""
+
+import sys
+
+from garak import _config
+from garak.probes.base import Probe
+
+this = sys.modules[__name__]
+
+def load_local_data(self):
+    pass
+
+def local_constructor(self):
+    super(self.__class__, self).__init__()
+    self.load_data()
+
+class DoNotAnswer_Probe(Probe):
+    """Do Not Answer probe
+
+    A curated list of questions provided in the scientific research
+    which are to be supplied as prompts and LLMs should not respond back."""
+
+    probe_tags = [
+        "discrimination_exclusion_toxicity_hateful_offensive",
+        "human_chatbox",
+        "information_hazard",
+        "malicious_uses",
+        "misinformation_harms",
+    ]
+    goal = "Check if the model answers to unwanted prompts"
+    
+    prompts = ["<|endoftext|>", ".", "<Links>", "<Wiki>", ""]
+
+    max_challenging = {}
+
+    for tag in probe_tags:
+        # load filter lists
+        with open(
+            _config.transient.basedir
+            / f"resources/donotanswer/{tag}.txt",
+            encoding="utf=8",
+        ) as f:
+            max_challenging[tag] = f.read().strip().split("\n")
+
+        # define class
+        classname = "DoNotAnswer_" + tag.title()
+        descr = f"Implementation of {tag} probes in DoNotAnswer_Prompts"
```

Comment:
If keeping the file names:
```suggestion
        classname = "DoNotAnswer_" + tag.title().replace("_", "")
        descr = f"Implementation of {classname} probes in DoNotAnswer_Prompts"
```
or
```suggestion
        classname = "DNA" + tag.title().replace("_", "")
        descr = f"Implementation of {classname} probes in DoNotAnswer_Prompts"
```
or
```suggestion
        classname = tag.title().replace("_", "")
        descr = f"Implementation of {classname} probes in DoNotAnswer_Prompts"
```